mitodl/mit-open
# Filled vs Unfilled Bookmarks

### What are the relevant tickets?
https://github.com/mitodl/hq/issues/4712

### Description (What does it do?)
Makes userlist bookmark and learningpath icon filled vs unfilled based on membership in at least 1 list.

### Screenshots (if appropriate):

**Cards**
<img width="874" alt="Screenshot 2024-06-26 at 6 32 13 PM" src="https://github.com/mitodl/mit-open/assets/9010790/72ea0e01-0209-4bd7-b171-3ed2978f6721">

**List Cards**
<img width="1041" alt="Screenshot 2024-06-26 at 6 32 46 PM" src="https://github.com/mitodl/mit-open/assets/9010790/2cc00d60-8f86-42f4-9b78-53dbac63bbec">

**List Cards, Mobile**
<img width="734" alt="Screenshot 2024-06-26 at 6 32 57 PM" src="https://github.com/mitodl/mit-open/assets/9010790/b976cd8e-88fb-4237-9086-801533ad5ad6">


### How can this be tested?

1. As a logged in user, click the bookmark button on a card to change its membership in "userlists".
    - try this throughout the app, particularly for horizontal "ListCard" (search page) and vertical cards
2. When the resource belongs to at least one list, the bookmark should be filled.
3. When the resource belongs to zero lists, the bookmark should be unfilled.
4. The stafflist button should behave similarly.
5. Try adding/removing featured courses (homepage "Featured Courses" carousel) to a **userlist**.
    - in contrast to RC, the carousel resources should maintain their order.
